numpy/core/src/multiarray/mapping.c

```diff
@@ -169,7 +169,7 @@ prepare_index(PyArrayObject *self, PyObject *index,
     int new_ndim, fancy_ndim, used_ndim, index_ndim;
     int curr_idx, get_idx;
 
-    npy_intp i, n;
+    int i, n;
```

Comment:
Yeah, n needs to be intp, but i is probably fine. Though would need to check more carefully that it is only used to iterate something like axes/dimensions.